### PR TITLE
OpenSSH audit: range tests.

### PR DESCRIPTION
**Pull request by a-martynovich**
#277

### CODE REVIEW

File: agent/security_helper.py [hunk 1]
```diff
@@ -163,19 +164,51 @@ def selinux_status():
...
+    @property
+    def safe_value(self) -> str:
+        return self._safe_value
+
+    def __init__(self, default, safe, compare=COMPARE.MATCH):
+        self.default = default
+        self._safe = safe
+        self.is_safe = {self.COMPARE.MATCH: self.match,
+                        self.COMPARE.RANGE: self.range}[compare]
+        self._safe_value = (safe if compare == self.COMPARE.MATCH else str(self._safe[1]))
```
Comment (rptrchv) [inline agent/security_helper.py]: Redundant parentheses

File: agent/__main__.py [hunk 2]
```diff
@@ -21,14 +22,37 @@ def main():
...
+        'openssh-empty-password':
+            ('OpenSSH: Disable logins with empty password', 'PermitEmptyPasswords'),
+        'openssh-root-login':
+            ('OpenSSH: Disable root login', 'PermitRootLogin'),
+        'openssh-password-auth':
+            ('OpenSSH: Disable password authentication', 'PasswordAuthentication'),
+        'openssh-agent-forwarding':
+            ('OpenSSH: Disable agent forwarding', 'AllowAgentForwarding'),
+        'openssh-protocol':
+            ('\tOpenSSH: Force protocol version 2', 'Protocol'),
```
Comment (rptrchv) [inline agent/__main__.py]: What's the purpose of this '\t' here?

File: agent/security_helper.py [hunk 3]
```diff
@@ -163,19 +164,51 @@ def selinux_status():
...
+        return vmin <= int(val) <= vmax
+
+    @property
+    def safe_value(self) -> str:
+        return self._safe_value
+
+    def __init__(self, default, safe, compare=COMPARE.MATCH):
+        self.default = default
+        self._safe = safe
+        self.is_safe = {self.COMPARE.MATCH: self.match,
```
Comment (rptrchv) [inline agent/security_helper.py]: Pls comment

File: agent/security_helper.py [hunk 4]
```diff
@@ -362,7 +395,7 @@ def patch_sshd_config(patch_param):
         logger.exception('sshd or service executable not found')
         return
 
-    safe_value_string = '\n# Added by wott-agent on {}\n{} {}\n'.format(time.ctime(), patch_param, safe_value)
+    safe_value_string = '\n# Added by wott-agent on {}\n{} {}\n'.format(time.ctime(), patch_param, param_info.safe_value)
```
Comment (rptrchv) [inline agent/security_helper.py]: Too long line

File: agent/security_helper.py [hunk 5]
```diff
@@ -382,15 +415,15 @@ def patch_sshd_config(patch_param):
...
+                    logger.info('{}: replacing "{}" with "{}"'.format(param, value, param_info.safe_value))
                     patched_lines[-1] = safe_value_string
                     replaced = True
                     safe = False
                 else:
                     safe = True
-        if not replaced and not safe and default_value != safe_value:
-            logger.info('{}: replacing default "{}" with "{}"'.format(patch_param, default_value, safe_value))
+        if not replaced and not safe and not param_info.is_safe(param_info.default):
+            logger.info('{}: replacing default "{}" with "{}"'.format(patch_param, param_info.default, param_info.safe_value))
```
Comment (rptrchv) [inline agent/security_helper.py]: Too long line

File: agent/__main__.py (hunk 2 above)
Comment (a-martynovich) [reply agent/__main__.py]: To align the output in the console. Otherwise it's hard to read.

**Comment by a-martynovich** [comment]
I'm doing this to silence Codacy warnings. It says i should always use %-formatting for logging. It is a valid point, though

**Pull request merged**